src/main/scala/io/citrine/lolo/bags/BaggedResult.scala

```diff
@@ -407,13 +408,47 @@ case class MultiPredictionBaggedResult(
   }
 }
 
+/** Enumerates the ways in which uncertainty correlation can be calculated. */
+object CorrelationMethods extends Enumeration {
+  type CorrelationMethod = Value
+
+  val Trivial, FromTraining, Bootstrap, Jackknife, JackknifeExplicit = Value
+}
+
 /**
   * Container with model-wise predictions for each label and the machinery to compute (co)variance.
   *
   * @param baggedPredictions  bagged prediction results for each label
   * @param realLabels         a boolean sequence indicating which labels are real-valued
+  * @param NibIn              the sampling matrix as (# bags) x (# training)
   */
-case class MultiTaskBaggedResult(baggedPredictions: Seq[BaggedResult[Any]], realLabels: Seq[Boolean]) extends BaggedResult[Seq[Any]] with MultiTaskModelPredictionResult {
+case class MultiTaskBaggedResult(
+                                  baggedPredictions: Seq[BaggedResult[Any]],
+                                  realLabels: Seq[Boolean],
+                                  NibIn: Vector[Vector[Int]],
+                                  trainingLabels: Seq[Seq[Any]],
+                                  trainingWeights: Seq[Double]
```

Comment:
`trainingLabels` and `trainingWeights` are being piped through so that the correlation over training data can be computed, but that method does not work very well and these arguments will almost certainly be removed by the time this code is released.

Reply:
It's non-local, so unlikely to work in any but the most contrived examples